virt-v2v/cold/entrypoint.go

```diff
@@ -127,6 +128,21 @@ func buildCommand() []string {
 				virtV2vArgs = append(virtV2vArgs, "--mac", macToIp)
 			}
 		}
+		// Adds LUKS keys, if exist.
+		if file, err := os.Open("/etc/luks/luks"); err == nil {
+			defer file.Close()
+
+			scanner := bufio.NewScanner(file)
+			for scanner.Scan() {
+				virtV2vArgs = append(virtV2vArgs, "--key", fmt.Sprintf("all:key:%s", scanner.Text()))
+			}
+			if err := scanner.Err(); err != nil {
+				fmt.Println("Error scanning the LUKS keys file")
+				panic(fmt.Errorf("error %w", err))
+			}
+		} else {
+			fmt.Println("Error opening the LUKS keys file ", err)
```

Comment:
but what if the file doesn't exist because LUKS keys were not specified?
I know I'm jumping between these two options, but could be simpler to set the data from the secret as environment variables..

Reply:
> we show this message in this case. if the vm disk is encrypted - the v2v pod will fail with the relevant error. otherwise it will continue as usual.

ok, I meant when LUKS keys are not specified and the VM doesn't have encrypted disks (which is the common case) - we shouldn't display this error then..

Reply:
ok, in that case either we don't show anything.
i see two options:
- if the disk with encryption, we will fail later. troubleshooting would be checking the command can tell if we had keys and it continues there..
- switching to environment var as you suggested

Reply:
i went with env var at the end